### PR TITLE
feat: `Activity` dataclass

### PR DESCRIPTION
**Pull request by juvilius**
and ci: dependency group instead of optional dependency for linting

I've had some troubles pleasing mypy, so please check if I've made a mistake before considering to merge. :)

Also: your `Data` class is focused on there being one of "it" per day. I'm not sure how to best reconcile it with activities.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -74,6 +74,13 @@ indent-style = "space"
 skip-magic-trailing-comma = false
 line-ending = "auto"
 
+[dependency-groups]
+linting = [
+    "mypy>=1.13.0",
+    "ruff>=0.8.2",
+    "types-requests>=2.32.0.20241016",
+]
```
[the remark this answers is not in the document]
Comment (matin) [reply pyproject.toml:82]: @juvilius what's the advantage of this vs the current approach?

**Review by matin — CHANGES_REQUESTED** [review]
@juvilius can you add tests?

**Comment by matin** [comment]
@juvilius I apologize for not getting around to reviewing earlier. do you want to rebase, add tests and submit the changes?

adding this endpoint makes sense, especially to make available in [`garth-mcp-server`](https://github.com/matin/garth-mcp-server/)

**Comment by vactomas** [comment]
Hi, @juvilius,
Is there anything I could help you with in order to get this PR ready? I could use this feature for my personal project as well.

**Comment by juvilius** [comment]
Hi @matin,

sorry for the holdup. Let me explain.
I had modified _garth_ enough to be able to generate a map of my running tracks. To not have to rely on Garmin, I stored everything in a postgres DB. I therefore had to create a data model, which turned out to be a bother with `Activities` coming in all different shapes and sizes depending on the type of activity and even on the device/watch. This lead me to lazily store everything as JSON and hardcode only a handful of _fields_.

Long story short, I don't know how to easily collect every possible field for all types of `activity` in a `dataclass`.
If anybody has an idea, go ahead :)

**Review by matin — APPROVED** [review]
